README.md

```diff
@@ -896,3 +897,145 @@ def add_item_to_cart(cart, item):
 
 **[⬆ back to top](#table-of-contents)**
 
+# **Dependency Inversion Principle (DIP)**
+The DIP amounts to the following: 
+- *Depend upon abstractions. Do not depend upon concrete classes.*
+
+The object of this principle is to help decouple software and render high-level modules independant from low-level module implementation details. In detail, the principle states:
+1. High-level modules should not depend on low-level modules. Both should depend on abstractions (e.g. interfaces).
+2. Abstractions should not depend on details. Details (concrete implementations) should depend on abstractions.
+
+By dictating that both high-level and low-level objects must depend on the same abstraction, this design principle inverts the way some people may think about object-oriented programming.
+
+Thinking about the interactions as abstract concepts themselves, allows us to de-couple components while maintaining a lighter and less implementation-dependant interaction schema.
+
+Sources for this module: [Wikipedia](https://en.wikipedia.org/wiki/Dependency_inversion_principle), [Head First Design Patterns](http://ce.sharif.edu/courses/96-97/2/ce484-1/resources/root/Design%20Patterns/Eric%20Freeman,%20Elisabeth%20Freeman,%20Kathy%20Sierra,%20Bert%20Bates-Head%20First%20Design%20Patterns%20-OReilly%20(2008).pdf)
+
+### **Mixins**
+Mixins are a special type of multiple inheritance. There are two typical situations that 
+require mixins:
+1. You want to add in a lot of features and/or methods for a class
+2. You want to use a particular feature and/or across many different classes
```

Comment:
I think an important point to note is that a `mixin` should not hold internal state. It should only add functionality via methods or properties

Reply:
Great point, I might add a tiny section or list of points titled, `what is a mixin?`.

Reply:
Or maybe, `what is not a mixin?`